File: main/src/main/java/com/sedmelluq/discord/lavaplayer/tools/http/HttpContextFilter.java
package com.sedmelluq.discord.lavaplayer.tools.http;

import org.apache.http.HttpResponse;
import org.apache.http.client.methods.HttpUriRequest;
import org.apache.http.client.protocol.HttpClientContext;

public interface HttpContextFilter {
    void onContextOpen(HttpClientContext context);

    void onContextClose(HttpClientContext context);

    void onRequest(HttpClientContext context, HttpUriRequest request, boolean isRepetition);

    boolean onRequestResponse(HttpClientContext context, HttpUriRequest request, HttpResponse response);

    boolean onRequestException(HttpClientContext context, HttpUriRequest request, Throwable error);
}

File: main/src/main/java/lavaplayer/tools/http/HttpContextFilter.kt
package lavaplayer.tools.http

import org.apache.http.HttpResponse
import org.apache.http.client.methods.HttpUriRequest
import org.apache.http.client.protocol.HttpClientContext

interface HttpContextFilter {
    fun onContextOpen(context: HttpClientContext)

    fun onContextClose(context: HttpClientContext)

    fun onRequest(context: HttpClientContext, request: HttpUriRequest, isRepetition: Boolean)

    fun onRequestResponse(context: HttpClientContext, request: HttpUriRequest, response: HttpResponse): Boolean

    fun onRequestException(context: HttpClientContext, request: HttpUriRequest, error: Throwable): Boolean
}
